Automattic/wp-calypso
# Hosting Metrics: Restore 'Auto refresh' functionality for logs

Fixes https://github.com/Automattic/dotcom-forge/issues/3388
See pdKhl6-2cB-p2#comment-3723

## Proposed Changes

Restores the 'Auto refresh' functionality for the log viewer. I placed it within `<SiteLogsToolbar />` to make it a setting specific to each page.

It's off by default, and disables if you select a date range or go to a new page. It also resets when you switch tabs.

### Desktop

<img width="1059" alt="CleanShot 2023-08-09 at 14 23 45@2x" src="https://github.com/Automattic/wp-calypso/assets/36432/57bdadb9-79a4-4e16-b06c-beb65e30f2fc">

### Mobile

<img width="531" alt="CleanShot 2023-08-09 at 14 25 38@2x" src="https://github.com/Automattic/wp-calypso/assets/36432/c328ccff-ef0e-42b3-8712-36614590477a">

## Testing Instructions

1. Navigate to `/site-monitoring/<sites>` and click on 'PHP logs'.
2. Verify auto refresh is disabled by default.
3. Enable auto refresh and verify the logs refresh after ~10 seconds.
4. Change the date range and verify auto refresh is disabled.
5. Click on 'Webserver logs' and repeat the same testing steps.
